src/core/tests/ov_tensor_test.cpp

```diff
@@ -42,7 +42,7 @@ TEST_F(OVTensorTest, canCreateTensor) {
     ASSERT_EQ(byteStrides(ov::Strides({6, 2, 1}), t.get_element_type()), t.get_strides());
     ASSERT_EQ(ov::element::f32.size() * totalSize, t.get_byte_size());
     ASSERT_THROW(t.data(ov::element::i64), ov::Exception);
-    ASSERT_THROW(t.data<std::int32_t>(), ov::Exception);
+    ASSERT_NO_THROW(t.data<std::int32_t>());
```

Comment:
do we need to allow f32 a int32?
should we allow only specific types of reinterpter casts?

Reply:
Added additional check to avoid cast from real to integral